internal/interaction/transaction.go

```diff
@@ -18,7 +18,30 @@ import (
 )
 
 func (s *serviceInteractor) GetTransactionsForDebitor(ctx context.Context, query entities.TransactionQuery) ([]entities.Transaction, error) {
-	return s.store.GetTransactionsByFilter(ctx, query)
+	logger := logging.LoggerFromContext(ctx)
+	mgr := NewIdentityManager(ctx)
+
+	if mgr.IsRegisteredUser() {
```

Comment:
this is true for admins also, right?

And for an admin token, the call to ListMyRegistrationIds will fail.

So should be
```
if mgr.IsAdmin() || mgr.IsAPITokenCall() {
  // ...
} **else** if mgr.IsRegisteredUser() {
  // ...
}
```

Reply:
Misread.